paddle/fluid/inference/tests/book/test_inference_machine_translation_decode.cc

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <gtest/gtest.h>
+#include "gflags/gflags.h"
+#include "paddle/fluid/inference/tests/test_helper.h"
+
+DEFINE_string(dirname, "", "Directory of the inference model.");
+
+// Test for testing inference using "decode_main" in python
+TEST(inference, machine_translation_decode) {
+  if (FLAGS_dirname.empty()) {
+    LOG(FATAL) << "Usage: ./example --dirname=path/to/your/model";
+  }
+
+  LOG(INFO) << "FLAGS_dirname: " << FLAGS_dirname << std::endl;
+  std::string dirname = FLAGS_dirname;
+
+  // 0. Call `paddle::framework::InitDevices()` initialize all the devices
+  // In unittests, this is done in paddle/testing/paddle_gtest_main.cc
+
+  // Setup init_ids and init_scores
+  paddle::framework::LoDTensor init_ids, init_scores;
+  paddle::framework::LoD init_lod{{0, 1}};
+  init_lod.push_back({0, 1});
+
+  std::vector<int64_t> init_ids_data = {1};
+  SetupLoDTensor<int64_t>(init_ids, {1, 1}, init_lod, init_ids_data);
+
+  std::vector<float> init_scores_data = {1};
+  SetupLoDTensor<float>(init_scores, {1, 1}, init_lod, init_scores_data);
+
+  // Setup input sequence of 5 words
+  paddle::framework::LoDTensor input_sequence;
+  std::vector<int64_t> inp_data = {100, 50, 8, 94, 122};
```

Comment:
Does this data have a specific meaning? Or can we use random data?

Reply:
This is a random sequence of 5 words (The ids need to be within a range).